api/spec/src/billing/customeroverride.tsp

```diff
@@ -0,0 +1,124 @@
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace OpenMeter.Billing;
+
+@route("/api/v1/billing/customer")
+@tag("Billing (Experimental)")
+interface CustomerOverrides {
+  /**
+   * List customer overrides
+   */
+  @get
+  @summary("List customer overrides")
+  @operationId("billingListCustomerOverrides")
+  list(
+    @query
+    billingProfile?: Array<ULID>,
+
+    ...OpenMeter.QueryPagination,
+    ...OpenMeter.QueryLimitOffset,
+    ...OpenMeter.QueryOrdering<CustomerOverrideOrderBy>,
+  ): OpenMeter.PaginatedResponse<CustomerOverride> | OpenMeter.CommonErrors;
+
+  /**
+   * Create/update a new customer override.
+   */
+  @post
+  @route("/{customerIdOrKey}")
+  @summary("Create/update a customer override")
+  @operationId("billingUpsertCustomerOverride")
+  upsert(
+    @path
+    customerIdOrKey: ULIDOrKey,
+
+    @body
+    request: CustomerWorkflowOverride,
+  ):
+    | {
+        // Updated
+        @statusCode _: 200;
+
+        @body body: CustomerOverride;
+      }
+    | {
+        // Created
+        @statusCode _: 201;
+
+        @body body: CustomerOverride;
+      }
+    | OpenMeter.NotFoundError
+    | OpenMeter.CommonErrors;
+
+  /**
+   * Get a customer override by id.
+   */
+  @get
+  @route("/{customerIdOrKey}")
+  @summary("Get a customer override")
+  @operationId("billingGetCustomerOverrideById")
+  get(
+    @path
+    customerIdOrKey: ULIDOrKey,
+  ): CustomerOverride | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Delete a customer override by id.
+   */
+  @delete
+  @route("/{customerIdOrKey}")
+  @summary("Delete a customer override")
+  @operationId("billdingDeleteCustomerOverride")
+  delete(
+    @path
+    customerIdOrKey: ULIDOrKey,
+  ): {
+    @statusCode _: 204;
+  } | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+}
+
+/**
+ * Order by options for customers.
+ */
+@friendlyName("BillingCustomerOverrideOrderBy")
+enum CustomerOverrideOrderBy {
+  id: "id",
+  createdAt: "created_at",
+  updatedAt: "updated_at",
```

Comment:
I think these timestamps are not indexed today. I wouldn't add them until we don't have UI use-case.